### PR TITLE
SLI-856 Rework open Hotspot in IDE to select detected one

### PR DESCRIPTION
**Pull request by henryju**

### CODE REVIEW

**Review by nquinquenel — APPROVED** [review]
LGTM

File: src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java [hunk 1]
```diff
@@ -217,22 +216,19 @@ public void showIssueLocations(LiveIssue liveIssue) {
...
 
-  public void show(LocalHotspot remoteHotspot) {
-    //TODO plug in when OPEN in IDE feature modified
-//    activeHotspot = liveHotspot;
-    if (getToolWindow() == null) {
-      // can happen if we try to show while the project is opening
-      // we will show the hotspot when the tool window is built
-      return;
+  public boolean trySelectSecurityHotspot(String securityHotspotKey) {
+    var foundHotspot = liveSecurityHotspots.stream().filter(sh -> securityHotspotKey.equals(sh.getServerFindingKey())).findFirst();
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java:220]: I noticed this from Mithil's branch, I'm not sure we should keep a list of live findings in the tool window (there is a logic to merge etc, I think it's too much for a UI component). In your case you could rely on the `OnTheFlyFindingsHolder` from `ShowSecurityHotspotCallable`.

File: src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java [hunk 2]
```diff
@@ -117,6 +118,12 @@ public void analyzeFilesOnUserAction(Collection<VirtualFile> files, AnActionEven
     }
   }
 
+  public void analyzeFileAndTrySelectHotspot(VirtualFile file, String securityHotspotKey) {
+    AnalysisCallback callback = new ShowSecurityHotspotCallable(project, onTheFlyFindingsHolder, securityHotspotKey);
+    var task = new Analysis(project, List.of(file), TriggerType.OPEN_SECURITY_HOTSPOT, true, callback);
+    TaskRunnerKt.startBackgroundTask(project, ANALYSIS_TASK_TITLE, task::run);
```
Comment (damien-urruty-sonarsource) [inline src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java:124]: I would have expected to have a modal task. As the user explicitly request to open the hotspot in the IDE, I think it's better to show him that something is going on

**Review by damien-urruty-sonarsource — APPROVED** [review]
2 comments else LGTM

File: src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java (hunk 1 above)
Comment (nquinquenel) [reply src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java:220]: This is something I removed on my refactoring, maybe I can take care of that on my rebase later?

Comment (henryju) [reply src/main/java/org/sonarlint/intellij/actions/SonarLintToolWindow.java:220]: OK, so I let you handle the change @nquinquenel

File: src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java (hunk 2 above)
Comment (henryju) [reply src/main/java/org/sonarlint/intellij/analysis/AnalysisSubmitter.java:124]: I will change that in a later PR, I don't think it is blocker for the merge.

**Pull request merged**